### DISCUSSION TITLE
seqcli ingest JSON data from plain text log statements

### DISCUSSION THREAD

**Discussion by mkvonarx** [datalust/seq-tickets#1536] [Ideas]
Hi

We're using serilog for our logging. Unfortunately, we cannot user the CLEF JSON format in production. Instead we use a more human readable format with # to delimit the values and a {Properties} at the end to also output all remaining Serilog properties. Our format string looks almost like this:
"outputTemplate": "{Timestamp:yyyy-MM-dd HH:mm:ss.fff} # {ThreadId,3} # {Level:u3} # {SourceContext,-30} # {Message:lj} # {Properties} # {Exception}${NewLine}"

We're trying to ingest these logs to Seq with "seqcli ingest -x" and it works mostly fine. The only thing we cannot manage to nicely ingest is the {Properties} part. This is a JSON inside the human readable text, and seqcli ingest does not have a JSON parser according to https://docs.datalust.co/docs/command-line-client#extraction-patterns

Is there any way to ingest the {Properties} with seqcli ingest as anything else than a string?

Having a 5th builtin property like e.g. @p that directly understands JSON like {Properties} (like @t etc.) would propably be the nicest solution. Or maybe something like "seqcli ingest -x ".... {:json} ..." could also work.

**Comment by nblumhardt** [comment]
Hi Markus, thanks for the note. I really like the `{:json}` direction - it'd be nice to support capturing a JSON value into a single property, or merging in properties from a JSON object into the top-level event.

I'll move this over to `datalust/seqcli` so we can track it alongside the code. I'll take a look at what JSON support in extraction patterns would look like, I don't think there's a quick and easy option given the functionality in `seqcli ingest` today 🤔

**Reply by nblumhardt** [reply]
Created: https://github.com/datalust/seqcli/issues/238
